src/components/editor/Editor.tsx

```diff
@@ -67,22 +99,27 @@ class Editor extends React.Component<Partial<IEditorProps>, Partial<IEditorState
   };
 
   static stateFromProps(props: IEditorProps): IEditorState {
-    // TODO perhaps replace AnnotationViewModel with AnnotationModel
-    const annotation = props.annotation || new AnnotationViewModel();
     return {
-      priority: annotation.priority || AnnotationPriorities.NORMAL,
-      comment: annotation.comment || '',
-      referenceLink: annotation.referenceLink || '',
+      ...props.form,
+      locationX: props.editor.locationX,
+      locationY: props.editor.locationY,
+      calculateInverted: props.calculateInverted,
+      isDragged: false,
       referenceLinkError: '',
-      referenceLinkTitle: annotation.referenceLinkTitle || '',
       referenceLinkTitleError: '',
       noCommentModalOpen: false,
     };
   }
 
+  moverElement: RefObject<HTMLDivElement>;
+  dragTracker: DragTracker;
+
   constructor(props: IEditorProps) {
     super(props);
+    console.log(props);
     this.state = Editor.stateFromProps(props);
```

Comment:
if the function was called `getDerivedStateFromProps`, then you wouldn't have to invoke it manually as it is part of the component lifecycle

https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
Done